code/modules/industrial_lift/elevator/elevator_machinery.dm

```diff
@@ -373,3 +389,385 @@
+	var/datum/lift_master/lift = lift_ref?.resolve()
+
+	// Check for stopped states.
+	if(!lift || !is_operational)
+		// Lift missing, or we lost power.
+		set_lift_state(0, 0, force = !is_operational)
+		return PROCESS_KILL
+
+	use_power(active_power_usage)
+
+	var/obj/structure/industrial_lift/lift_part = lift.lift_platforms[1]
+
+	if(QDELETED(lift_part))
+		set_lift_state(0, 0, force = !is_operational)
+		return PROCESS_KILL
+
+	// Update
+	set_lift_state(current_lift_direction, lift.lift_platforms[1].z - lowest_floor_offset)
+
+	// Lift's not moving, we're done; we just had to update the floor number one last time.
+	if(!current_lift_direction)
+		return PROCESS_KILL
+
+/**
+ * Set the state and update appearance.
+ *
+ * Arguments:
+ * new_direction - new arrow state: UP, DOWN, or 0
+ * new_floor - set the floor number, eg. 1, 2, 3
+ * force_update - force appearance to update even if state didn't change.
+ */
+/obj/machinery/lift_indicator/proc/set_lift_state(new_direction, new_floor, force = FALSE)
+	if(new_direction == current_lift_direction && new_floor == current_lift_floor && !force)
+		return
+
+	current_lift_direction = new_direction
+	current_lift_floor = new_floor
+	update_appearance()
+
+/obj/machinery/lift_indicator/update_appearance(updates)
+	. = ..()
+
+	if(!is_operational)
+		set_light(l_on = FALSE)
+		maptext = ""
+		return
+
+	set_light(l_on = TRUE)
+	maptext = {"<div style="font:5pt 'Small Fonts';color:[LIGHT_COLOR_DARK_BLUE]">[current_lift_floor]</div>"}
+
+/obj/machinery/lift_indicator/update_overlays()
+	. = ..()
+
+	if(!is_operational)
+		return
+
+	. += emissive_appearance(icon, "[base_icon_state]e", offset_spokesman = src, alpha = src.alpha)
+
+	if(!current_lift_direction)
+		return
+
+	var/arrow_icon_state = "[base_icon_state][current_lift_direction == UP ? "up" : "down"]"
+
+	. += mutable_appearance(icon, arrow_icon_state)
+	. += emissive_appearance(icon, "[arrow_icon_state]e", offset_spokesman = src, alpha = src.alpha)
+
+MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/lift_indicator, 32)
+
+/obj/machinery/door/window/elevator
+	name = "elevator door"
+	desc = "Keeps idiots like you from walking into an open elevator shaft."
+	can_atmos_pass = ATMOS_PASS_DENSITY // elevator shaft is airtight when closed
+	req_access = list(ACCESS_TCOMMS)
+	var/id = null
+	var/safety_enabled = TRUE
+
+/obj/machinery/door/window/elevator/Initialize(mapload, set_dir, unres_sides)
+	. = ..()
+	RemoveElement(/datum/element/atmos_sensitive, mapload)
+	GLOB.elevator_doors += src
+
+/obj/machinery/door/window/elevator/Destroy()
+	GLOB.elevator_doors -= src
+	return ..()
+
+/obj/machinery/door/window/elevator/bumpopen(mob/user)
+	if(operating || !density)
+		return
+	add_fingerprint(user)
+	if(!requiresID())
+		user = null
+	if(!safety_enabled || allowed(user))
+		open_and_close()
+	else
+		do_animate("deny")
+	return
+
+/obj/machinery/door/window/elevator/proc/cycle_doors(command)
+	switch(command)
+		if(OPEN_DOORS)
+			if(density)
+				open()
```

Comment:
is this the same `open()` that calls `set_density()`? `set_density()` has an early return baked in (dependent to the `density`) so there's no need to check it all the way up here imo. you can keep it if you want but there's no true reason... worst case scenario just remove the density check here and comment something close to what i just said (if it's true)

Reply:
It was just windoors missing it, so I fixed that.